core/src/main/scala/caliban/codegen/ScalaWriter.scala

```diff
@@ -0,0 +1,256 @@
+package caliban.codegen
+
+import caliban.parsing.adt.{ Document, ExecutableDefinition, Selection, Type }
+import caliban.parsing.adt.Type.{ FieldDefinition, ListType, NamedType }
+import Generator._
+import caliban.parsing.adt.ExecutableDefinition.{ FragmentDefinition, OperationDefinition, TypeDefinition }
+import caliban.parsing.adt.Selection.{ Field, FragmentSpread, InlineFragment }
+
+object ScalaWriter {
+  trait ScalaGQLWriter extends GQLWriterContext {
+    implicit val fieldWriter            = FieldWriter
+    implicit val typeWriter             = TypeWriter
+    implicit val typeDefWriter          = TypeDefinitionWriter
+    implicit val docWriter              = DocumentWriter
+    implicit val rootQueryWriter        = RootQueryDefWriter
+    implicit val queryWriter            = QueryDefWriter
+    implicit val rootMutationWriter     = RootMutationDefWriter
+    implicit val mutationWriter         = MutationDefWriter
+    implicit val rootSubscriptionWriter = RootSubscriptionDefWriter
+    implicit val subscriptionWriter     = SubscriptionDefWriter
+    implicit val fragmentWriter         = FragmentWriter
+    implicit val selectionWriter        = SelectionTypeWriter
+    implicit val argsWriter             = ArgsWriter
+  }
+
+  object DefaultGQLWriter extends ScalaGQLWriter
+
+  //todo hmm
+//  case class CaseClass(name: String, ) {
+  //  toString
+  //  }
+
+  def reservedType(typeDefinition: TypeDefinition): Boolean =
+    typeDefinition.name == "Query" || typeDefinition.name == "Mutation" || typeDefinition.name == "Subscription"
+
+  object DocumentWriter extends GQLWriter[Document, Any] {
+    override def write(schema: Document)(nothing: Any)(implicit context: GQLWriterContext): String = {
+      import context._
+
+      val hasSubscriptions = Document.typeDefinition("Subscription")(schema).nonEmpty
+
+      s"""import Types._
+      import Fragments._
+      ${if (hasSubscriptions) "import zio.console.Console\n      import zio.stream.ZStream" else ""}
```

Comment:
The type for subscription is 
`A => ZStream[Console, Nothing, %s]`

Reply:
I think it should be `Any` instead of `Console`. Maybe you got the `Console` from one of the examples, but in practice we don't know what users will need in their ZIO environment so we can just use `Any`.